src/Client/Updater.php

```diff
@@ -634,4 +638,51 @@ public function download(string $target, ...$extra): PromiseInterface
         return $this->repoFileFetcher->fetchTarget($target, $length, ...$extra)
             ->then($verify);
     }
+
+    /**
+     * @param string $target
+     *   The path of the target file. Needs to be known to the most recent
+     *   targets metadata downloaded in ::refresh().
+     */
+    private function getMetadataForTarget(string $target, SnapshotMetadata $snapshotMetadata, ?TargetsMetadata $targetsMetadata = null): ?TargetsMetadata
+    {
+        static $searchedRoles = [];
```

Comment:
I know static properties are the same for all instances of a class, but how does it work for static variables in class methods? I just want to be certain this won't get polluted by other instances of `Updater` running in the same memory space (i.e., what the Composer plugin does).

Reply:
Yes, this would have been a problem. No longer using static vars here. Good catch.